### PR TITLE
[Access-tokens] Add image from github.com/mm4tt/k8s-util/experiments

### PR DESCRIPTION
**Pull request by jprzychodzen**
This image is required to run clusterloader2 load test with access-tokens testing

Bug: kubernetes/kubernetes#83375

### CODE REVIEW

**Comment by mm4tt** [comment]
Some remarks:

1. You don't need both vendor and go modules, I'd recommend having just go modules (would be much easier to review :))
2. I don' think this should be a top-level directory in a cluster loader. We should put it into some intermediate directory as we might have more such "tools" (utils? testutils?) in the future

/ok-to-test

**Comment by jprzychodzen** [comment]
Ad #1 - Vendor is required, as otherwise Travis presubmit is broken. As discussed, I will change Travis presubmit to skip testing this code and I will add new Prow presubmit for this image.
Ad #2 - Sure, make sense.

File: .travis.yml [hunk 1]
```diff
@@ -28,5 +28,5 @@ script:
     - ./verify/verify-flags-underscore.py
     - ./verify/verify-dashboard-format.sh
     - go get -d ./...
-    - travis_wait go build $(go list ./... | grep -v "/vendor/")
```
Comment (mm4tt) [inline .travis.yml]: Could you extract this change to a separate PR?

Comment (jprzychodzen) [reply .travis.yml]: Done, as #996

**Comment by jprzychodzen** [comment]
/hold
Waiting for #996

File: util-images/access-tokens/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Access-tokens
+
+Utility image used in access token testing. Generate constant number of requests per second, based on provided paramters.
```
Comment (mm4tt) [inline util-images/access-tokens/README.md]: * Mention https://github.com/kubernetes/kubernetes/issues/83375, it contains a write-up on what "access token testing" means
* s/Generate/Generates/ or maybe something like "Utility can be used to generate X QPS to kube-apiserver using Y access tokens where both X, Y are configurable via flags"

File: util-images/access-tokens/example/example.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  name: access-tokens
+subjects:
+  - kind: ServiceAccount
+    name: access-tokens
+    namespace: access-tokens
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: access-tokens
```
Comment (mm4tt) [inline util-images/access-tokens/example/example.yaml]: This is a single token, so access-token?

File: verify/verify-flags/known-flags.txt [hunk 4]
```diff
@@ -1,4 +1,5 @@
 apiserver-address
+access-token-dirs
```
Comment (mm4tt) [inline verify/verify-flags/known-flags.txt]: It's a tragedy that we need to update it every time we add a flag...
I opened https://github.com/kubernetes/perf-tests/issues/997 to address it

File: util-images/access-tokens/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,16 @@
+FROM golang:1.13.4 AS build-env
+
+ARG gopkg=k8s.io/perf-tests/access-tokens
```
[the remark this answers is not in the document]
Comment (jprzychodzen) [reply util-images/access-tokens/Dockerfile]: Updated.

File: util-images/access-tokens/example/example.yaml (hunk 3 above)
Comment (jprzychodzen) [reply util-images/access-tokens/example/example.yaml]: Name for all resources is consistent and delivered from name of the utility (access-tokens).

If this name will be changed, then all other resources also need to have more specific names (like service-account-getter for a Role).

I will try to update all names into more descriptive form.

File: util-images/access-tokens/Makefile [hunk 6]
```diff
@@ -0,0 +1,17 @@
+PROJECT = k8s-testimages
+IMG = gcr.io/$(PROJECT)/access-tokens
```
[the remark this answers is not in the document]
Comment (jprzychodzen) [reply util-images/access-tokens/Makefile]: Added additional directory in path, changed to `gcr.io/$(PROJECT)/perf-tests-util/access-tokens`

File: verify/verify-flags/known-flags.txt (hunk 4 above)
Comment (jprzychodzen) [reply verify/verify-flags/known-flags.txt]: Great, thanks!

File: util-images/access-tokens/README.md (hunk 2 above)
Comment (jprzychodzen) [reply util-images/access-tokens/README.md]: Done. Also, updated other parts of readme to support copy-paste version to test and run.

**Comment by jprzychodzen** [comment]
/hold

I want to confirm that Prow presubmit works with this change.

File: util-images/access-tokens/cmd/main.go [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+		return nil, rest.ErrNotInCluster
+	}
+
+	token, err := ioutil.ReadFile(tokenFile)
+	if err != nil {
+		return nil, err
+	}
+
+	tlsClientConfig := rest.TLSClientConfig{}
+
```
Comment (mm4tt) [inline util-images/access-tokens/cmd/main.go]: NIT: Delete blank line(s)

File: util-images/access-tokens/cmd/main.go [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/klog"
+)
+
+var (
+	namespace = flag.String("namespace", "", "Namespace the pod is running in")
```
Comment (mm4tt) [inline util-images/access-tokens/cmd/main.go]: NIT: "the pod" is a bit ambiguous here. Maybe something more like "Namespace where the pod running this binary is running"?

File: util-images/presubmit.sh [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+set -u
+
+DIRNAME=$(dirname "$(readlink -f "$0")")
+
+TARGETS=("access-tokens")
+
+for target in "${TARGETS[@]}"; do
+  cd "${DIRNAME}/${target}"
+  make build
+done
```
Comment (mm4tt) [inline util-images/presubmit.sh]: NIT: Newline

File: util-images/access-tokens/cmd/main.go (hunk 8 above)
Comment (jprzychodzen) [reply util-images/access-tokens/cmd/main.go]: Changed to `Namespace that is targeted with request flow.` to describe exactly what is happening.

File: util-images/access-tokens/cmd/main.go (hunk 7 above)
Comment (jprzychodzen) [reply util-images/access-tokens/cmd/main.go]: Done

File: util-images/presubmit.sh (hunk 9 above)
Comment (jprzychodzen) [reply util-images/presubmit.sh]: Done

**Comment by mm4tt** [comment]
Looks like it works - https://prow.k8s.io/?job=pull-perf-tests-util-images
Let's change the `skip_report` in the presubmit to `false` to have it reported on github

**Comment by mm4tt** [comment]
/lgtm

Thanks!
